test/fast/FastDistributionsFacet.test.ts

```diff
@@ -1,29 +1,249 @@
 import * as chai from "chai";
+import { expect } from "chai";
+import { BigNumber } from "ethers";
 import { solidity } from "ethereum-waffle";
-import { smock } from "@defi-wonderland/smock";
-
+import { FakeContract, MockContract, smock } from "@defi-wonderland/smock";
+import { deployments, ethers } from "hardhat";
+import { SignerWithAddress } from "hardhat-deploy-ethers/signers";
+import { Issuer, FastDistributionsFacet, Marketplace, Fast, FastAccessFacet, IERC20, Distribution } from "../../typechain";
+import { fastFixtureFunc } from "../fixtures/fast";
+import { abiStructToObj, DistributionPhase } from "../utils";
 chai.use(solidity);
 chai.use(smock.matchers);
 
 describe("FastDistributionsFacet", () => {
-  before(async () => { });
-  beforeEach(async () => { });
+  let deployer: SignerWithAddress,
+    issuerMember: SignerWithAddress,
+    governor: SignerWithAddress,
+    alice: SignerWithAddress;
+
+  let issuer: FakeContract<Issuer>,
+    marketplace: FakeContract<Marketplace>,
+    fast: Fast,
+    accessMock: MockContract<FastAccessFacet>,
+    erc20: FakeContract<IERC20>,
+    distributions: FastDistributionsFacet,
+    governorDistributions: FastDistributionsFacet;
+
+  const fastDeployFixture = deployments.createFixture(fastFixtureFunc);
+
+  const resetIssuerMock = () => {
+    issuer.isMember.reset();
+    issuer.isMember.whenCalledWith(issuerMember.address).returns(true);
+    issuer.isMember.returns(false);
+  };
+
+  // Setup for a successful distribution.
+  const setupSuccessfulDistribution = async () => {
```

Comment:
Super clean! Well done. However - I would suggest having this function created in the scope of the `describe` block that has tests which uses it - so that it doesn't pollute the top-level.